### PR TITLE
Code reference bypass

### PR DESCRIPTION
**Pull request by HardNorth**
1. Moved requirements into a separate file, since some code scan tools (E.g. fossa.com) don't have idea about different requirements declaration methods. Well, actually this point is discussable.
2. Added 'codeRef' parameter bypass and some unit-tests

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -131,3 +131,70 @@ def test_get_system_information_without_pkg(self):
         cond = (ReportPortalService.get_system_information('pytest')
                 == expected_result)
         assert cond
+
+    @mock.patch("reportportal_client.service._get_data",
```
Comment (iivanou) [inline tests/test_service.py:135]: Please, use either single quotes or double ones within the same module.

File: tests/test_service.py [hunk 2]
```diff
@@ -131,3 +131,70 @@ def test_get_system_information_without_pkg(self):
...
+        "field_name,field_value,expected_name,expected_value",
+        start_item_optional)
+    @mock.patch("reportportal_client.service._get_data",
+                mock.Mock(return_value={"id": 123}))
+    def test_start_item_code_optional_params(self, rp_service, field_name,
+                                             field_value, expected_name,
+                                             expected_value):
+        """Test for validate different fields in start_test_item.
+
+        :param: rp_service: fixture of ReportPortal
```
Comment (iivanou) [inline tests/test_service.py:176]: Please, align param descriptions to the same level.

File: tests/test_service.py [hunk 3]
```diff
@@ -131,3 +131,70 @@ def test_get_system_information_without_pkg(self):
...
+        :param: field_name a name of a field bypassed to
```
Comment (iivanou) [inline tests/test_service.py:177]: :param param_name: description

File: tests/test_service.py [hunk 4]
```diff
@@ -131,3 +131,70 @@ def test_get_system_information_without_pkg(self):
...
+                                     'type': 'STORY', 'parameters': None,
+                                     'hasStats': True,
+                                     'codeRef': None},
+                               url='http://endpoint/api/v2/project/item',
+                               verify=True)
+
+        rp_service.session.post.assert_called_with(**expected_result)
+        assert rp_start == 123
+
+    start_item_optional = [
```
Comment (iivanou) [inline tests/test_service.py:159]: class attributes need to be defined before methods.

File: tests/test_service.py (hunk 2 above)
Comment (HardNorth) [reply tests/test_service.py:176]: I tried (see commit history), but pydocstyle breaks build in that case: `D208: Docstring is over-indented`.
BTW do we really need that 79 characters line length limitation, in 2020?

File: tests/test_service.py (hunk 4 above)
Comment (HardNorth) [reply tests/test_service.py:159]: If we talk about regular class then yes, but this is unit-tests. I just try too keep data near a location where it is used.

File: tests/test_service.py (hunk 2 above)
Comment (iivanou) [reply tests/test_service.py:176]: You can use line breaks for long descriptions. 79 is good, 120 is used mostly in UI testing.

Comment (HardNorth) [reply tests/test_service.py:176]: Ok, got what you mean.

**Pull request merged**